src/main/java/org/folio/service/caches/JobProfileSnapshotCache.java

```diff
@@ -0,0 +1,72 @@
+package org.folio.service.caches;
+
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.json.Json;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.rest.jaxrs.model.ProfileSnapshotWrapper;
+import org.folio.rest.util.OkapiConnectionParams;
+import org.folio.rest.util.RestUtil;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import org.apache.http.HttpStatus;
+
+import com.github.benmanes.caffeine.cache.AsyncCache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+@Component
+public class JobProfileSnapshotCache {
+
+  private static final Logger LOGGER = LogManager.getLogger();
+
+  @Value("${orders.profile-snapshot-cache.expiration.time.seconds:3600}")
+  private long cacheExpirationTime;
+  private AsyncCache<String, Optional<ProfileSnapshotWrapper>> cache;
```

Comment:
I think it is fine to return optional in the api methods of this class. Having optional saved into the cache means that it is possible to put Optional.empty into the cache. Is that intentional?

Reply:
we will fix it during using this cache

Reply:
from aliaks, `we will use this code in nearest future and we will be able to add your remark the moment we use this code. At this point, the task was to prepare the code so we could quickly start adding the necessary business logic.`